### PR TITLE
Dev #T864: Sticky Bar

### PR DESCRIPTION
**Pull request by gabrieljenik**
- Get sticky topbar in on SETTINGS of survey on all 'Survey settings' and 'Survey menu'
- If easy also get sticky topbar in all all pages which have topbar

### CODE REVIEW

File: application/helpers/LayoutHelper.php [hunk 1]
```diff
@@ -445,6 +445,11 @@ protected function beforeSideMenuRender(array $aData)
      */
     public static function renderTopbar(array $aData)
     {
+        App()->getClientScript()->registerScriptFile(
+            App()->getConfig('adminscripts') . 'topbar.js',
```
Comment (Shnoulle) [inline application/helpers/LayoutHelper.php:449]: Question : why we don't use assetmanager for adminscripts ?

**Comment by gabrieljenik** [comment]
The automated test is failing. 
```
Question Group Editor (ls\tests\QuestionGroupEditor)
 ✘ Edit question group
   │
   │ Facebook\WebDriver\Exception\UnrecognizedExceptionException: Element <a href="#editgrp_de"> is not clickable at point (635,78) because another element <div id="surveybarid" class="menubar surveybar sticky"> obscures it
```

Seems there is some overlapping on the "testing" screen view. Is there any screenshot we can look?
Maybe saved at tests/tmp/screenshots? Although no sure if it is kept.

**Comment by olleharstedt** [comment]
!    │ Facebook\WebDriver\Exception\NoSuchElementException: Unable to locate element: #execute_survey_button

Did this button change?

**Comment by olleharstedt** [comment]
> Maybe saved at tests/tmp/screenshots? Although no sure if it is kept.

No, they're only saved when running tests locally. It's usually just easier to do the test manually step by step, imitating the code in the test class.

**Comment by gabrieljenik** [comment]
I think we are ready to go! automated testing is green

File: application/helpers/LayoutHelper.php (hunk 1 above)
Comment (olleharstedt) [reply application/helpers/LayoutHelper.php:453]: You mean a new package? @Shnoulle

File: assets/scripts/admin/topbar.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+$(document).on('ready pjax:scriptcomplete', function () {
+    function makeTopbarSticky() {
+        const topbar = $('.menubar:not(.surveymanagerbar)');
+        if (topbar.length == 0) {
+            return;
+        }
+        const topbarOffset = topbar.offset().top;
+
+        // When the topbar is floated, the space it occupies is released, so the rest of the content "jumps" up.
```
Comment (olleharstedt) [inline assets/scripts/admin/topbar.js:9]: Nice with comments. :+1:

**Comment by olleharstedt** [comment]
Good stuff, guys!

File: application/helpers/LayoutHelper.php (hunk 1 above)
Comment (Shnoulle) [reply application/helpers/LayoutHelper.php:453]: No package needed : it's a question and i really thing it's always better to use assetmanager.

`App()->getClientScript()->registerScriptFile(App()->assetManager->publish(App()->getConfig('adminscripts')) . '/' . 'topbar.js');`

For a quick way

Comment (olleharstedt) [reply application/helpers/LayoutHelper.php:453]: It's not automatically applied for regsiterScriptFile()?

Comment (olleharstedt) [reply application/helpers/LayoutHelper.php:453]: See source in application/core/LSYii_ClientScript.php

Comment (Shnoulle) [reply application/helpers/LayoutHelper.php:453]: Arg ! But it's not the best ! It create different asset for all files ! 

Must confirm : but i think it create a copy of all `adminscripts` for each file !

Comment (olleharstedt) [reply application/helpers/LayoutHelper.php:453]: OK, well, let me know. :)

Comment (Shnoulle) [reply application/helpers/LayoutHelper.php:453]: And report as issue …

File: assets/scripts/admin/topbar.js (hunk 2 above)
Comment (gabrieljenik) [reply assets/scripts/admin/topbar.js:9]: I like comments :) Sometimes more important than code.
Code reflects what I think is the solution.
Comment tells what I think is the solution.
If code fails, comments allow to still see which was the intention

**Pull request merged**